### DISCUSSION TITLE
Will 3D-Grid be imported to 3d-pannel in the future?

### DISCUSSION THREAD

**Discussion by AndyYangjd** [foxglove/community#35] [Q&A]
I want to have a 3d grid box which can set the inverval and length.

Below is the photo from [varyar](https://www.vayyar.com.cn/technology.html)
![image](https://user-images.githubusercontent.com/44262434/176568533-567484ff-dd0a-485a-ae76-7518c6e206e8.png)

**Comment by defunctzombie** [comment]
The 3d panel supports [cube markers](https://foxglove.dev/docs/studio/panels/3d#marker). These markers can have any color and opacity you want so you can already accomplish this today.

**Comment by AndyYangjd** [comment]
I means whether the default grid in 3d-pannel can be extended to three dimension. Also, the new 3d-grid have the some config to  current 2d-grid.

**Comment by jhurliman** [comment]
We chatted on Slack, my understanding of the request is about adding a “3D Grid” to the layout, similar to how 2D Grids are added to the layout.

@AndyYangjd I’m looking T that screenshot; is it right that this is not a dense 3D grid? It looks like one large 2D grid on the ground plane, and three smaller 2D grids forming three sides of a cube?

If that’s the case, you can make this today by setting the rotation angle of three 2D grids, and using one larger 2D grid for the ground.

Note that the currently released 3D panel beta has a few bugs in the 2D Grid settings, you’ll likely see rendering issues until that’s fixed.

**Reply by AndyYangjd** [reply]
> my understanding of the request is about adding a “3D Grid” to the layout, similar to how 2D Grids are added to the layout.

Yeah, that's my request. Because now i can't find the exact photo to display this request, so i add the screenshot.

I just want a easy way to see all the points in the space of 3d-pannel.
Besides, the `cube marker` can realize the ui in the above screenshot, but it will cost some time to do it.

**Comment by jhurliman** [comment]
Here is an example of creating a scene similar to your screenshot in the `3D (Beta)` panel.

<img width="1312" alt="Screen Shot 2022-07-07 at 11 57 06 AM" src="https://user-images.githubusercontent.com/195374/177869956-72e0f0de-255a-4b63-8406-79f8ff61e6b7.png">

**Reply by AndyYangjd** [reply]
Yeah, you'r right. But it contains five grids. Maybe we only need the size and subdivisions/interval value  of x, y, z. 6 params can do it
![微信图片_20220711185111](https://user-images.githubusercontent.com/44262434/178248631-0b5ce5ed-ce10-4cab-adec-11f600d21965.jpg)
.
